### PR TITLE
Add wmts service in geo document_builder

### LINKED ISSUE #4898: ScannedMaps with raster resources should have IIIF and mosaic web service endpoints.
To support new mosaic workflow.

Close this ticket when MapSet records have the mosaic web service indexed.

### PR DESCRIPTION
**Pull request by thanyabegum**
Closes #4898

### CODE REVIEW

File: app/services/geo_discovery/document_builder/wxs.rb [hunk 1]
```diff
@@ -81,6 +81,15 @@ def generate_wms_path?
           @config && visibility && file_set && (raster_file_set? || vector_file_set?)
         end
 
+        # Determines if the wmts path should be generated
+        # @return [Boolean]
+        def generate_wmts_path?
+          return false unless visibility == "open"
+          return true if raster_set?
+          return true if raster_file_set?
+          resource_decorator.object.decorate.try(:mosaic_file_sets).present?
```
Comment (tpendragon) [inline app/services/geo_discovery/document_builder/wxs.rb:90]: Suggestion: Add `Wayfinder.for(resource).try(:mosaic_file_set_count)`, and then make that method use `custom_queries.find_deep_children_with_property(resource: resource, model: FileSet, property: :service_targets, value: "mosaic", count: true)` - that should be faster and more stable than serializing all the file sets into objects.

**Review by tpendragon — CHANGES_REQUESTED** [review]
Just one suggested change. Open to discussion on it too.

File: app/services/geo_discovery/document_builder/wxs.rb (hunk 1 above)
Comment (hackartisan) [reply app/services/geo_discovery/document_builder/wxs.rb:90]: Sounds good. Do we want to just change the one we added and continue going through the resource decorator? Or we aren't trying to contain use of wayfinders? There's a lot of using them via the resource decorators.

Comment (tpendragon) [reply app/services/geo_discovery/document_builder/wxs.rb:90]: Yeah that's a good call, let's continue using the decorator.

**Review by eliotjordan — DISMISSED** [review]

File: app/resources/scanned_maps/scanned_map_decorator.rb [hunk 2]
```diff
@@ -50,6 +50,7 @@ def attachable_objects
            :geo_metadata_members,
            :members,
            :parents,
+           :mosaic_file_sets,
```
[the remark this answers is not in the document]
Comment (hackartisan) [reply app/resources/scanned_maps/scanned_map_decorator.rb:53]: @eliotjordan the builder just goes ahead and adds the service for anything that's a raster resource or raster fileset. Is that not the correct behavior?

[the remark this answers is not in the document]
Comment (hackartisan) [reply app/resources/scanned_maps/scanned_map_decorator.rb:53]: We had a discussion about this, and whether the RasterSet case needs to also check for service type. @eliotjordan will create a new issue.

**Comment by hackartisan** [comment]
superceded by #4958 and #4962

**Pull request closed without merging**